packages/toolbar/src/browser/application-shell-with-toolbar-override.ts

```diff
@@ -60,12 +61,14 @@ export class ApplicationShellWithToolbarOverride extends ApplicationShell {
     protected tryShowToolbar(): boolean {
         const doShowToolbarFromPreference = this.toolbarPreferences[TOOLBAR_ENABLE_PREFERENCE_ID];
         const isShellMaximized = this.mainPanel.hasClass(MAXIMIZED_CLASS) || this.bottomPanel.hasClass(MAXIMIZED_CLASS);
-        if (doShowToolbarFromPreference && !isShellMaximized) {
+        const show = doShowToolbarFromPreference && !isShellMaximized;
+        if (show) {
             this.toolbar.show();
-            return true;
+        } else {
+            this.toolbar.hide();
         }
-        this.toolbar.hide();
-        return false;
+        setActualTopHeightForContentHoverWidget(show ? 32 * 2 : 32);
```

Comment:
Could you elaborate where the 32 comes from? In the browser I see that the top panel has 32px but the toolbar only has a height of 30px. Overall I'm a bit unsure about those hard-coded numbers. Wouldn't it be better to use something like `this.topPanel.node.getBoundingClientRect().height` and `this.toolbar.node.getBoundingClientRect().height` instead?

Reply:
> Could you elaborate where the 32 comes from? In the browser I see that the top panel has 32px but the toolbar only has a height of 30px.

Thank you! I updated the PR. No ~32~ (which was incorrect) magic number, but the height is calculated from the toolbar + topPanel's bound rects.

I resolve this thread; please comment and reopen it if you disagree.